tests/karma/core/components/TestPaginate.js

```diff
@@ -1,123 +1,159 @@
+          assert.equal(link.childNodes.length, 1);
+          assert.equal(link.childNodes[0].tagName, 'A');
+          assert.equal(link.textContent, 'hi');
+        });
       });
     });
   });
 
-  describe('makeLink()', () => {
-    let pager;
-    let root;
-
-    beforeEach(() => {
-      pager = sinon.spy();
-      root = new Paginate({count: 50, currentPage: 5, pager});
-    });
-
-    describe('when the link is to the current page', () => {
-      it('does not contain a link', () => {
-        const link = renderIntoDocument(root.makeLink({currentPage: 3, page: 3, pager}));
-        assert.equal(link.childNodes.length, 1);
-        assert.equal(link.childNodes[0].nodeType, Node.TEXT_NODE);
-        assert.equal(link.textContent, '3');
-      });
-
-      it('uses the provided text', () => {
-        const link = renderIntoDocument(
-          root.makeLink({currentPage: 3, page: 3, pager, text: 'hi'}));
-        assert.equal(link.childNodes.length, 1);
-        assert.equal(link.childNodes[0].nodeType, Node.TEXT_NODE);
-        assert.equal(link.textContent, 'hi');
+  describe('links', () => {
+    const pathname = '/some-path/';
+
+    class PaginateWrapper extends React.Component {
+      render() {
+        return <Paginate count={250} currentPage={5} pathname={pathname} showPages={5} />;
+      }
+    }
+
+    function renderPaginate() {
+      return new Promise((resolve) => {
+        const node = document.createElement('div');
+        render((
+          <Router history={createMemoryHistory('/')}>
+            <Route path="/" component={PaginateWrapper} />
+          </Router>
+        ), node, () => {
+          resolve(node);
+        });
       });
-    });
-
-    describe('when the link is to a different page', () => {
-      it('has a link', () => {
-        const link = renderIntoDocument(root.makeLink({currentPage: 2, page: 3, pager}));
-        assert.equal(link.childNodes.length, 1);
-        assert.equal(link.childNodes[0].tagName, 'A');
-        assert.equal(link.textContent, '3');
-      });
-
-      it('uses the provided text', () => {
-        const link = renderIntoDocument(
-          root.makeLink({currentPage: 4, page: 3, pager, text: 'hi'}));
-        assert.equal(link.childNodes.length, 1);
-        assert.equal(link.childNodes[0].tagName, 'A');
-        assert.equal(link.textContent, 'hi');
-      });
-
-      it('calls the pager when clicked', () => {
-        assert(!pager.called, 'pager was called early');
-        const link = renderIntoDocument(
-          root.makeLink({currentPage: 4, page: 3, pager, text: 'hi'}));
-        Simulate.click(link.querySelector('a'));
-        assert(pager.calledWith(3));
+    }
+
+    it('renders the right links', () => {
+      renderPaginate().then((root) => {
+        const links = root.querySelectorAll('a');
+        assert.deepEqual(
+          Array.prototype.map.call(
+            links, (link) => [link.textContent, link.getAttribute('href')]),
```

Comment:
I've mostly been using `Array.from` (at least after I wrote this I discovered it and started using it) so I converted it to use that.